### PR TITLE
LED subsystem

### PR DESCRIPTION
**Pull request by jagernet-ops**

### CODE REVIEW

File: src/main/java/frc/robot/notification/LEDStateEnum.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package frc.robot.notification;
+
+public enum LEDStateEnum {
+  COLOR_RED_SPARK(0.61, "Disabled"), 
+  COLOR_YELLOW_SPARK(0.69, "Updating"), 
+  COLOR_GREEN_SPARK(0.77, "Enabled");
```
Comment (python-42) [inline src/main/java/frc/robot/notification/LEDStateEnum.java:6]: These names could just be the color, I dont think we need the COLOR_*_SPARK

File: src/main/java/frc/robot/notification/LEDSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  private final PWMSparkMax ledController = new PWMSparkMax(Constants.LEDController.LED_CONTROLLER_CHANNEL); 
+  private LEDStateEnum status;
+
+  public LEDSubsystem(){
+    tab = Shuffleboard.getTab("LEDStatus");
+    tab.addBoolean("Connection", () -> ledController.isAlive());
+    tab.addString("Status:", () -> status.getStatusMessage());
+  }
+
+  public void handleStateUpdate(LEDStateEnum status){
```
Comment (python-42) [inline src/main/java/frc/robot/notification/LEDSubsystem.java:20]: Could we just call this `changeColor()` or `changeStatus`

File: src/main/java/frc/robot/notification/ChangeColor.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  @Override
+  public void execute(){
+
+  }
+
+  @Override
+  public void end(boolean interrupeted){
+
+  }
```
Comment (python-42) [inline src/main/java/frc/robot/notification/ChangeColor.java:25]: Should explicitly declare `isFinished()` (return true). Could delete end and execute since they do nothing

File: src/main/java/frc/robot/notification/ChangeColor.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package frc.robot.notification;
+
+import edu.wpi.first.wpilibj2.command.Command;
+
+public class ChangeColor extends Command {
+  private LEDSubsystem subystem;
+  
+  public ChangeColor(){
+    subystem = new LEDSubsystem();
```
Comment (python-42) [inline src/main/java/frc/robot/notification/ChangeColor.java:9]: Need to pass in subsystem can't make a new one over and over

**Review by python-42 — CHANGES_REQUESTED** [review]
Looks good other than a couple of things.

**Review by python-42 — APPROVED** [review]
Nice work Noah and Michael. We should be able to test this next time the whole electrical team is in the shop.

**Comment by python-42** [comment]
I have a question: Are we supposed to be using a PWMSparkMax rather than a regular SparkMax?
@jagernet-ops @NoahWeishan

File: src/main/java/frc/robot/notification/ILedIO.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package frc.robot.notification;
+
+public interface ILedIO {
+  
+  public void setSpark(double sparkValue);
```
Comment (pordonj) [inline src/main/java/frc/robot/notification/ILedIO.java:5]: We should comminucate to the hardware abstraction to change the color, not set its motor controller (the hardware anstraction can worry about how it sets the color and how to switch to a given color)

File: src/main/java/frc/robot/notification/SimLedIO.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+public class SimLedIO implements ILedIO{
+
+  private double ledSparkValue = 0;
+  
+  public SimLedIO(){}
+
+  @Override
+  public void setSpark(double sparkValue){
+    ledSparkValue = sparkValue;
```
Comment (pordonj) [inline src/main/java/frc/robot/notification/SimLedIO.java:11]: If we do the above comment then this can make a `Mechanism2d` on the Sim GUI and change its color to show for the simulator

File: src/main/java/frc/robot/notification/ILedIO.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+
+  public boolean getIsAlive();
```
Comment (pordonj) [inline src/main/java/frc/robot/notification/ILedIO.java:7]: is this relevant to the operation of the robot? Or just for debugging? Can we come up with examples of when this would be the data point that solved a problem?

**Review by pordonj — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/notification/ILedIO.java (hunk 7 above)
Comment (jagernet-ops) [reply src/main/java/frc/robot/notification/ILedIO.java:7]: The boolean method is useful for if we determine there to be a connection issue with the blinkin'. When running the simulator we just return true, but when enabled this will return the connection status of the blinkin' over the PWM channel.

**Review by pordonj — APPROVED** [review]
